torchx/tracker/api.py

```diff
@@ -246,7 +248,7 @@ def run_from_env() -> AppRun:
         this method will retorn same AppRun and hence same instances of the backend implementations.
         """
 
-        torchx_job_id = os.environ["TORCHX_JOB_ID"]
+        torchx_job_id = os.getenv("TORCHX_JOB_ID", default=Path(sys.argv[0]).name)
```

Comment:
In any multi-user environment this seems problematic since most jobs will have the job ID `python` right? Is it better to randomly generate an ID here instead? or require users to manually specify one if not present? Seems like a sharp edge

I think adding a new `run_id: Optional[str]` param to run_from_env() makes more sense than just defaulting it

Reply:
the issue with adding the `run_id` param to `run_from_env()` is that then it makes the app a bit confusing to read. Presumably I have something like this now in my code:

```
def my_main() -> None:
    app_run tracker.app_run_from_env("my-custom-run")
```

If I run the above as `python my_main.py` then the job-id is `"my-custom-run"` but if I launch it via torchx it becomes something like `local_cwd://torchx/utils-python_my_main-19ksjkc`

Reply:
I'd say the manual specified one should always take precedence

You could also construct it directly and not use env variables. But still defaulting the job id to "python" seems wrong

Reply:
per our offline discussion, updated the job_id to `"<UNDEFINED>"` when the job is not launched from torchx since there is no canonical job_id.